core/src/com/fallenflame/game/GameEngine.java

```diff
@@ -1,5 +1,384 @@
+     */
+    public GameCanvas getCanvas(){return canvas;}
+
+    /** Set the new GameCanvas*/
+    public void setCanvas(GameCanvas canvas){this.canvas = canvas;}
+
+    /**
+     * Creates a new game world
+     *
+     * The physics bounds and drawing scale are now stored in the levelController and
+     * defined by the appropriate JSON file.
+     *
+     * Game does not start out as active or paused, but rather in loading
+     *
+     * Source: Professor White
+     */
+    public GameEngine() {
+        jsonReader = new JsonReader();
+        level = new LevelController();
+        isSuccess = false;
+        isFailed = false;
+        isScreenActive = false;
+        isPaused = false;
+    }
+
+    /**
+     * Dispose of all (non-static) resources allocated to this mode.
+     */
+    public void dispose() {
+        level.dispose();
+        level  = null;
+        canvas = null;
+    }
+
+    /**
+     * Resets the status of the game so that we can play again.
+     *
+     * This method disposes of the level and creates a new one. It will
+     * reread from the JSON file, allowing us to make changes on the fly.
+     */
+    public void reset() {
+        level.dispose();
+
+        setComplete(false);
+        setFailure(false);
+        countdown = -1;
+
+        // Reload the json each time
+        String currentLevelPath = "jsons/" + saveJson.getString("current");
+        levelJson = jsonReader.parse(Gdx.files.internal("jsons/level.json"));
+        level.populate(levelJson);
+        level.getWorld().setContactListener(this);
+    }
+
+    /**
+     * Returns whether to process the update loop
+     *
+     * At the start of the update loop, we check if it is time
+     * to switch to a new game mode.  If not, the update proceeds
+     * normally.
+     *
+     * @param delta Number of seconds since last animation frame
+     *
+     * @return whether to process the update loop
+     */
+    public boolean preUpdate(float dt) {
+        InputController input = InputController.getInstance();
+        input.readInput();
+        if (listener == null) {
+            return true;
+        }
+
+        if (input.didDebug()) {
+            level.setDebug(!level.getDebug());
+        }
+
+        if (input.didReset()) {
+            reset();
+        }
+
+        if (input.didExit()) {
+            listener.exitScreen(this, EXIT_QUIT);
+            return false;
+        }
+
+        return true;
+    }
+
+    private Vector2 tempAngle = new Vector2();
+    /**
+     * The core gameplay loop of this world. This checks if the level has ended
+     * @param delta Number of seconds since last animation frame
+     */
+    public void update(float delta) {
+        InputController input = InputController.getInstance();
+
+        if (input.didFlare()) {
+            level.createFlare(input.getMousePosition());
+        }
+        if(input.didLight()){
+            level.lightFromPlayer(input.getLightRadius(), levelJson);
+        }
```

Comment:
Why is levelJson also passed in here? This is unnecessary I think.

Reply:
Wasn't sure if you would need it. Can def remove